tests/unit_tests/scheduler/test_scheduler.py

```diff
@@ -380,6 +380,44 @@ async def init(iens, *args, **kwargs):
     assert not sch.is_active()
 
 
+@pytest.mark.timeout(6)
+async def test_that_failed_realization_will_not_be_cancelled(realization, mock_driver):
+    started = asyncio.Event()
+    kill_called = False
+
+    async def wait(iens):
+        started.set()
+        # the job failed with exit code 1
+        return 1
```

Comment:
Shall I maybe parametrize this to check this both for success and failure? @berland @jonathan-eq

Reply:
Do we have another test that asserts jobs aren't cancelled on success?

Reply:
No, this one will now test both when 1) job completes then kill is not called 2) job didn't finish kill is called

Reply:
Are we testing if cancel is called or kill is called? We already have an issue with two separate PRs testing that kill is not called for completed (success and failed) realizations https://github.com/equinor/ert/issues/7658